src/controllers/dynakube/dynakube_controller.go

```diff
@@ -238,7 +238,7 @@ func (controller *Controller) reconcileDynaKube(ctx context.Context, dynakube *d
 		dynatraceClient,
 		controller.fs,
 		controller.versionProvider,
-		timeprovider.New(),
+		timeprovider.New().SetToNow(),
```

Comment:
Same here. All instances were the timeprovider is used, IMHO benefits of an more accurate now.

Reply:
`versionReconciler` calls 4 `versionStatusUpdaters`. IMHO it's better to call all of them or none of them (`timeProvider` is `Set`)  than call 1 updater out of 4 this time and 3 updaters out of 4 next time because the timeout expires at different points in time during the execution of the [loop](https://github.com/Dynatrace/dynatrace-operator/blob/main/src/controllers/dynakube/version/reconciler.go#L90)